### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by aikanshagarwal**

### CODE REVIEW

File: src/main/java/com/example/springassignment3/controller/TestController.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    @GetMapping("/tests/{testId}")
+    public Optional<Test> getTestById(@PathVariable int theId)
+    {
+        return theTestService.findTestById(theId);
+    }
+
+    @PostMapping("/tests")
+    public Test saveTest(@RequestBody Test theTest)
+    {
+        theTest.setId(0);
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/controller/TestController.java:33]: Add logs so that it will be easier to check if any exception occurs and see if happy path is working fine

File: src/main/java/com/example/springassignment3/controller/TestController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    @PutMapping("/tests")
+    public Test updateTest(@RequestBody Test theTest)
+    {
+        theTestService.saveTest(theTest);
+        return theTest;
+    }
+
+    @DeleteMapping("/tests/{testId}")
+    public void deleteTestById(@PathVariable int theId)
+    {
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/controller/TestController.java:47]: Make sure you use pathvariables and RequestParam

File: src/main/java/com/example/springassignment3/entity/Lab.java [hunk 3]
```diff
@@ -1,31 +0,0 @@
...
-    @Column(name = "name")
-    private String name;
-
-    @Column(name = "address")
-    private String address;
-
-    @OneToMany(mappedBy = "lab",cascade = CascadeType.ALL)
-    private List<Test> tests;
-
-}
```
Comment (tarunya882) [inline src/main/java/com/example/springassignment3/entity/Lab.java:31]: Why did you remove Lab entity?

Comment (aikanshagarwal) [reply src/main/java/com/example/springassignment3/entity/Lab.java:31]: It wasn't going to be used , will integrate the lab name in the test entity itself

File: src/main/java/com/example/springassignment3/controller/TestController.java (hunk 2 above)
Comment (aikanshagarwal) [reply src/main/java/com/example/springassignment3/controller/TestController.java:47]: Sure

File: src/main/java/com/example/springassignment3/controller/TestController.java (hunk 1 above)
Comment (aikanshagarwal) [reply src/main/java/com/example/springassignment3/controller/TestController.java:33]: Yes , sure
